apps/explorer/lib/explorer/chain/celo/election_reward.ex

```diff
@@ -69,4 +117,351 @@ defmodule Explorer.Chain.Celo.ElectionReward do
+
+  ## Returns
+  - A map where the keys are reward type atoms and the values are token atoms.
+
+  ## Examples
+
+      iex> ElectionReward.reward_type_atom_to_token_atom()
+      %{voter: :celo_token, validator: :usd_token, group: :usd_token, delegated_payment: :usd_token}
+  """
+  @spec reward_type_atom_to_token_atom() :: %{type => atom()}
+  def reward_type_atom_to_token_atom, do: @reward_type_atom_to_token_atom
+
+  @doc """
+  Builds a query to aggregate rewards by type for a given block hash.
+
+  ## Parameters
+  - `block_hash` (`Hash.Full.t()`): The block hash to filter rewards.
+
+  ## Returns
+  - An Ecto query.
+  """
+  @spec block_hash_to_aggregated_rewards_by_type_query(Hash.Full.t()) :: Ecto.Query.t()
+  def block_hash_to_aggregated_rewards_by_type_query(block_hash) do
+    from(
+      r in __MODULE__,
+      where: r.block_hash == ^block_hash,
+      select: {r.type, sum(r.amount), count(r)},
+      group_by: r.type
+    )
+  end
+
+  @doc """
+  Builds a query to get rewards by type for a given block hash.
+
+  ## Parameters
+  - `block_hash` (`Hash.Full.t()`): The block hash to filter rewards.
+  - `reward_type` (`type`): The type of reward to filter.
+
+  ## Returns
+  - An Ecto query.
+  """
+  @spec block_hash_to_rewards_by_type_query(Hash.Full.t(), type) :: Ecto.Query.t()
+  def block_hash_to_rewards_by_type_query(block_hash, reward_type) do
+    from(
+      r in __MODULE__,
+      where: r.block_hash == ^block_hash and r.type == ^reward_type,
+      select: r,
+      order_by: [
+        desc: :amount,
+        asc: :account_address_hash,
+        asc: :associated_account_address_hash
+      ]
+    )
+  end
+
+  @doc """
+  Builds a query to get rewards by account address hash.
+  """
+  @spec address_hash_to_rewards_query(Hash.Address.t()) :: Ecto.Query.t()
+  def address_hash_to_rewards_query(address_hash) do
+    from(
+      r in __MODULE__,
+      where: r.account_address_hash == ^address_hash,
+      select: r
+    )
+  end
+
+  @doc """
+  Builds a query to get ordered rewards by account address hash.
+
+  ## Parameters
+  - `address_hash` (`Hash.Address.t()`): The account address hash to filter
+    rewards.
+
+  ## Returns
+  - An Ecto query.
+  """
+  @spec address_hash_to_ordered_rewards_query(Hash.Address.t()) :: Ecto.Query.t()
+  def address_hash_to_ordered_rewards_query(address_hash) do
+    from(
+      r in __MODULE__,
+      join: b in assoc(r, :block),
+      as: :block,
+      preload: [block: b],
+      where: r.account_address_hash == ^address_hash,
+      select: r,
+      order_by: [
+        desc: b.number,
+        desc: r.amount,
+        asc: r.associated_account_address_hash,
+        asc: r.type
+      ]
+    )
+  end
+
+  @doc """
+  Makes Explorer.PagingOptions map for election rewards.
+  """
+  @spec block_paging_options(map()) ::
+          [
+            {:paging_options, Explorer.PagingOptions.t()},
+            ...
+          ]
+          | Explorer.PagingOptions.t()
```

Comment:
```suggestion
  @spec address_paging_options(map()) :: [Chain.paging_options()]
```

Don't forget to add Explorer.Chain alias

Reply:
Thanks! Fixed.